integration-tests/src/test/java/com/datastax/oss/driver/api/core/cql/ExecutionInfoWarningsIT.java

```diff
@@ -0,0 +1,224 @@
+    // verify the log was generated
+    Mockito.verify(appender, after(500).times(1)).doAppend(loggingEventCaptor.capture());
+    assertThat(loggingEventCaptor.getValue().getMessage()).isNotNull();
+    String logMessage = loggingEventCaptor.getValue().getFormattedMessage();
+    assertThat(logMessage)
+        .startsWith(
+            "Query '[0 values] "
+                + query
+                + "' generated server side warning(s): Aggregation query used without partition key");
+  }
+
+  @Test
+  public void should_execute_query_and_not_log_server_side_warnings() {
+    final String query = "SELECT count(*) FROM test;";
+    Statement<?> st =
+        SimpleStatement.builder(String.format(query))
+            .withExecutionProfileName("log-disabled")
+            .build();
+    ResultSet result = SESSION_RULE.session().execute(st);
+
+    ExecutionInfo executionInfo = result.getExecutionInfo();
+    assertThat(executionInfo).isNotNull();
+    List<String> warnings = executionInfo.getWarnings();
+    assertThat(warnings).isNotNull();
+    assertThat(warnings).isNotEmpty();
+    String warning = warnings.get(0);
+    assertThat(warning).isNotNull();
+    assertThat(warning).isEqualTo("Aggregation query used without partition key");
+    // verify the log was NOT generated
+    Mockito.verify(appender, after(500).times(0)).doAppend(loggingEventCaptor.capture());
+  }
+
+  @Test
+  public void should_expose_warnings_on_execution_info() {
+    // the default batch size warn threshold is 5 * 1024 bytes, but after CASSANDRA-10876 there must
+    // be
+    // multiple mutations in a batch to trigger this warning so the batch includes 2 different
+    // inserts.
+    final String query =
+        String.format(
+            "BEGIN UNLOGGED BATCH\n"
+                + "INSERT INTO test (k, v) VALUES (1, '%s')\n"
+                + "INSERT INTO test (k, v) VALUES (2, '%s')\n"
+                + "APPLY BATCH",
+            Strings.repeat("1", 2 * 1024), Strings.repeat("1", 3 * 1024));
+    Statement<?> st = SimpleStatement.builder(String.format(query)).build();
+    ResultSet result = SESSION_RULE.session().execute(st);
+    ExecutionInfo executionInfo = result.getExecutionInfo();
+    assertThat(executionInfo).isNotNull();
+    List<String> warnings = executionInfo.getWarnings();
+    assertThat(warnings).isNotNull();
+    assertThat(warnings).isNotEmpty();
+    // verify the log was generated
+    Mockito.verify(appender, after(500).times(1)).doAppend(loggingEventCaptor.capture());
+    assertThat(loggingEventCaptor.getValue().getMessage()).isNotNull();
+    String logMessage = loggingEventCaptor.getValue().getFormattedMessage();
+    assertThat(logMessage)
+        .startsWith("Query '")
+        // query will only be logged up to DefaultDriverOption.REQUEST_LOGGER_MAX_QUERY_LENGTH
+        // characters
+        .contains(
+            query.substring(
+                0,
+                SESSION_RULE
```

Comment:
Maybe we car replace this with a constant `500` since we now that that's the value we set above?